### ISSUE TITLE
Lint/Debugger offenses not found inside lambda assigned to field

### ISSUE THREAD

**Issue by ronocod** [rubocop/rubocop#12005]
## Expected behavior

RuboCop catches Lint/Debugger offenses across all code in a file.

## Actual behavior

Offenses inside a lambda which is being assigned to a field on an object are not found.

## Steps to reproduce the problem

In the following file, Rubocop finds a Lint/Debugger offense on line 3, but none on line 4:

```ruby
# frozen_string_literal: true

binding.pry
x.y = -> { binding.pry }
```

In this file, it's able to find both offenses:
```ruby
# frozen_string_literal: true

binding.pry
x = -> { binding.pry }
```

## RuboCop version

```
1.53.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.3) [arm64-darwin22]
```
